File: tools_java/tools/src/tools/CategoryInformation.java
package tools;

import java.io.BufferedReader;
import java.io.InputStream;
import java.io.InputStreamReader;
import java.nio.file.Files;
import java.nio.file.Path;
import java.nio.file.Paths;
/**
 * 
 * @author broihier
 * <pre>
 * Class for creating category information objects
 * </pre>
 */
public class CategoryInformation {
		private String categoryDescriptionText = "";
		private String categoryTitle = "";
		/** 
		 * Constructor for creating category information objects
		 * 
		 * <pre>
		 * Pseudo-code:
		 * {@code
		 * open the testDbCategoryTitle file and read the category title;
		 * open the testDbCategoryDescription file and read the category category description;
		 * 
		 * }
		 * </pre>
		 * @param category name that maps to 'category'.testDbCategoryTitle/Description files
		 * 
		 */
		public CategoryInformation(String category) {
			Path filePath = Paths.get(category+".testDbCategoryTitle");
			InputStream in = null;
			try {
				in = Files.newInputStream (filePath);
			} catch (Exception e) {
				System.out.println("Input stream creation error: " + e);
		 	}
			if (in != null) {
				BufferedReader reader = new BufferedReader(new InputStreamReader(in));
				try {
					String line = null;
					if ((line = reader.readLine()) != null) {
						categoryTitle = line.trim();						
					}
				} catch (Exception e) {
					System.out.println("Error while reading Category title database: "+e);
				}
			}
			filePath = Paths.get(category+".testDbCategoryDescription");
			try {
				in = Files.newInputStream (filePath);
			} catch (Exception e) {
				System.out.println("Input stream creation error: " + e);
		 	}
			if (in != null) {
				BufferedReader reader = new BufferedReader(new InputStreamReader(in));
				try {
					String line = null;
					if ((line = reader.readLine()) != null) {
						categoryDescriptionText = line.trim();						
					}
				} catch (Exception e) {
					System.out.println("Error while reading Category description database: "+e);
				}
			}
		}
		/**
		 * Method to get the category title
		 * @return categoryTitle - getter
		 */
		public String getCategoryTitle() {
			return categoryTitle;
		}
		/**
		 * Method to get the category description
		 * @return categoryDescriptionText - getter
		 */
		public String getCategoryDescriptionText() {
			return categoryDescriptionText;
		}

}

File: tools_C++/src/CategoryInformation.cpp
/*
 * CategoryInformation.cpp
 *
 *  Created on: Apr 12, 2017
 *      Author: broihier
 */

#include "CategoryInformation.h"
#include <stdio.h>
#include <iostream>
#include <fstream>

CategoryInformation::CategoryInformation(std::string category) {
	std::ifstream * DbFile = new std::ifstream();
	DbFile->open(category + ".testDbCategoryTitle");
	if (!DbFile->is_open()) {
		std::cout << "file open error: " + category + ".testDbCategoryTitle"
				<< std::endl;
		exit(-1);
	}
	char buffer[32767];
	DbFile->getline(buffer, sizeof(buffer));
	if (DbFile->gcount() > 0 && DbFile->gcount() < sizeof(buffer)) {
		categoryTitle = std::string(buffer);
	}
	DbFile->close();

	DbFile->open(category + ".testDbCategoryDescription");
	if (!DbFile->is_open()) {
		std::cout << "file open error: " + category + ".testDbCategoryDescription"
				<< std::endl;
		exit(-1);
	}
	DbFile->getline(buffer, sizeof(buffer));
	if (DbFile->gcount() > 0 && DbFile->gcount() < sizeof(buffer)) {
		categoryDescriptionText = std::string(buffer);
	}
	DbFile->close();
	delete DbFile;
}

CategoryInformation::~CategoryInformation() {
}

const std::string& CategoryInformation::getCategoryTitle() const{
	return categoryTitle;
}

const std::string& CategoryInformation::getCategoryDescriptionText() const{
	return categoryDescriptionText;
}
